source/fundamentals/connection/csot.txt

```diff
@@ -0,0 +1,225 @@
+.. _node-csot:
+
+Limit Server Execution Time
+===========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: error, blocking, thread, task
+
+Overview
+--------
+
+When you use {+driver-short+} to perform a server operation, you can also limit
+the duration allowed for the server to finish the operation. To do so,
+specify a **client-side operation timeout**. The timeout applies to all steps
+needed to complete the operation, including server selection, connection
+checkout, serialization, and server-side execution. When the timeout expires,
+{+driver-short+} raises a timeout exception.
+
+You can specify a timeout by using the ``timeoutMS`` option.
+
+.. note:: Experimental Feature
+
+   The Client-Side Operation Timeout (CSOT) feature is experimental and might
+   change in future driver releases.
+
+timeoutMS Option
+----------------
+
+You can specify the ``timeoutMS`` option in your ``MongoClientOptions``
+instance or at the database, collection, session, transaction, or operation
+levels.
+
+To specify a timeout when connecting to a MongoDB deployment, set the
+``timeoutMS`` connection option to the timeout length in milliseconds.
+
+The following code example uses the ``timeoutMS`` option to specify a timeout of
+10 seconds when instantiating a new ``MongoClient`` instance:
+
+.. code-block:: javascript
+
+   const uri = "mongodb://<db_username>:<db_password>@<hostname:<port>";
```

Comment:
S: remove authentication details since there are several kinds they could be using

```suggestion
   const uri = "mongodb://<hostname:<port>";
```